File: examples/apps/go/msgpacketizer-go/pkg/packetizer-go/encoder.go
package packetizer

import (
	"github.com/dim13/cobs"
	"github.com/sirupsen/logrus"
)

func Encode(index uint8, data []byte) []byte {
	logrus.Debugf("encode input: index=%d, data=%s", index, ToHex(data))
	var packet []byte
	packet = append(packet, index)
	packet = append(packet, data...)
	CRC8_HASH.Reset()
	CRC8_HASH.Update(data)
	crc8 := CRC8_HASH.CRC8()
	packet = append(packet, crc8)
	encoded := cobs.Encode(packet)
	logrus.Debugf("encoded: %s", ToHex(encoded))
	return append(encoded, COBS_DELIMITER)
}

File: examples/apps/python/msgpacketizer-py/src/packetizerpy/packetizer/encoder.py
import logging

import crc8
from cobs import cobs

from .common import COBS_DELIMITER, to_hex


def encode(index: int, data: bytes) -> bytes:
    logging.debug(f"encode input: {index}, {data}")
    hash = crc8.crc8()
    hash.update(data)
    digest = hash.digest()
    raw_data = bytes([index]) + data + digest
    encoded = cobs.encode(raw_data) + COBS_DELIMITER
    logging.debug(f"encoded: {to_hex(encoded)}")
    return encoded
